include/alpaka/mem/buf/BufUniformCudaHipRt.hpp

```diff
@@ -401,7 +401,7 @@ namespace alpaka
 
 
                         void * memPtr = nullptr;
-                        std::size_t pitchBytes = widthBytes;
+                        std::size_t pitchBytes = 0u;
```

Comment:
`pitchBytes` is overwritten with the real value later so there is no need to initialize it with a best guess.

Reply:
Please revert this change. I think I fixed this in the past in the case the followed if body is not accessed the pitch would be zero. This is wrong.

Reply:
Ah, now i also remember something like that.

Reply:
Ohh wait this is fine becaus ethe memory pointer points to `nullptr`

Reply:
Why does the pitch have to be !=0 when the size of the allocation is 0?
For the 3D case we set the pitch to zero as well in the exact same case.

Reply:
It was a wrong thinking on my side. All is fine with this change, if size is zero the pitch can be zero too.